src/examples/Tours/working_files/tour_from_linestring.py

```diff
@@ -46,8 +46,8 @@
     tour_doc[gxns+"Tour"].Playlist.append(flyto)
 
 assert Schema('kml22gx.xsd').validate(tour_doc)
-print etree.tostring(tour_doc, pretty_print=True)
+print(etree.tostring(tour_doc, pretty_print=True).decode())
 
 # output a KML file (named based on the Python script)
-outfile = file(__file__.rstrip('.py')+'.kml','w')
-outfile.write(etree.tostring(tour_doc, pretty_print=True))
+with open(__file__.rstrip('.py')+'.kml','wb') as outfile:
```

Comment:
Why is b (binary) being used for writing a text file?

Reply:
A KML is defined as being a stream of bytes, not a text file. The encoding in the KML header defines what is the encoding to be used (for example UTF-8).

See for example the following explanation in https://lxml.de/FAQ.html:

> ** Why can't lxml parse my XML from unicode strings?**
> First of all, XML is explicitly defined as a stream of bytes. It's not Unicode text. Take a look at the XML specification, it's all about byte sequences and how to map them to text and structure. That leads to rule number one: do not decode your XML data yourself. That's a part of the work of an XML parser, and it does it very well. Just pass it your data as a plain byte stream, it will always do the right thing, by specification.
> 
> This also includes not opening XML files in text mode. Make sure you always use binary mode, or, even better, pass the file path into lxml's parse() function to let it do the file opening, reading and closing itself. This is the most simple and most efficient way to do it.

In that example, we could save as text and it will work the same. I can change it if you want.

Reply:
BTW the etree.tostring in fact returns "bytes sream". 
It is just that in Python2 those were "string", and no it is expliciteky named "bytes" in Pytghon3/